### ISSUE TITLE
Менеджер SPI флешек

### ISSUE THREAD

**Issue by hedger** [flipperdevices/flipper-hackathon-moscow#3]
### Вы капитан команды?

- [x] Yes, I am a leader of my team

### Вы готовы приехать на Хакатон в Москву?

- [X] Yes, my team will attend

### Подробно опишите идею вашего проекта

Хотел бы сделать модуль для флиппера, суть такова...  

Пользователь может читать и шить SPI-флешки через прищепку или запаяв на макетку, вставляющуюся в гребёнку.
Опционально выполнять верификацию записанного, сравнивая с оригинальным файлом. 
Также сделать отображение параметров подключённой флешки - размер, вендор, ...
Поддерживаются флешки с уровнями 3.3 вольта. 

В составе модуля на флиппере будет простой менеджер файлов дампов на SD-карте.

Если с реализацией базового функционала не будет затыков, попробуем добавить связку с приложением под андроид для управления образами.

Модуль после доработки до приличного состояния можно будет включить в стоковую прошивку, т.к. функционал востребованный.

P.S. Второй год жду флиппер

### Готовый проект будет уметь...

Читать SPI флешки, сохранять образ на SD
Писать SPI флешки с образа на SD
Определять параметры подключённой флешки
Иметь интеграцию в UI флиппера
* Выбор операции и файла 
* Отображение параметров флешки
* Прогресс операции

### GitHub второго участника

@ghettorce

### GitHub третьего участника

@x893

**Comment by zhovner** [comment]
Предполагается ли GUI на самом экранчике флиппера для этой программы?

**Comment by hedger** [comment]
Да, для отображения параметров флешки, выбора образа, вывода прогресса операции.

**Comment by anna-oake** [comment]
Пожалуйста, добавьте про GUI для управления всеми этими фичами в раздел про критерии законченности.
Идея отличная!

**Comment by LionZXY** [comment]
Очень крутая идея!

**Comment by zhovner** [comment]
Хочется такую штуку для сброса пароля UEFI на макбуках https://www.youtube.com/watch?v=DrRzyLFdj9o

**Comment by hedger** [comment]
> Хочется такую штуку для сброса пароля UEFI на макбуках https://www.youtube.com/watch?v=DrRzyLFdj9o

На видео мелькает SCL/SDA - это ж I2C, но если будет рабочий каркас UI с функционалом под SPI, можно будет его тоже вкрутить

**Comment by x893** [comment]
Как можно присоединиться ?

**Comment by hedger** [comment]
> Как можно присоединиться ?

Написал в телегу

**Comment by hedger** [comment]
Апдейт: в команде +1, возможно, вытянем I2C в рамках хакатона. Возможности прикидываем

**Comment by x893** [comment]
Была информаця, что можно прислать firmware.bin для проверки на флиппере. Я сделал текущий dev (NO_BOOTLOADER) под Segger Embedded Studio (windows) и хотел проверить перед внесением дополнений. Можно загрузить прошвку и сказать - работает или нет ?

**Comment by zhovner** [comment]
@x893 спросите в чате пожалуйта. Сразу в одном сообщинии приложите прошивку и напишите что делать для проверки.

**Comment by alexey-gamov** [comment]
В качестве реферанса можно глянуть **XHorse chip programmer**

**Comment by hedger** [comment]
Зарегистрировался на Timepad

**Comment by yobabyte** [comment]
Зарегистрировался на Timepad

**Comment by x893** [comment]
Регистрируюсь
